### PR TITLE
Bluetooth: Audio: Shell: Fix metadata argument

### PR DESCRIPTION
**Pull request by Thalley**
The metadata in the presets are just defined
using the BT_AUDIO_LC3_PRESET which ultimately
uses BT_CODEC_LC3_CONFIG_META which correctly
sets the data pointer, but the `value` array is unused.

So we cannot just update the `value`, as that is actually not used in those cases, and the `data` pointer is `const`.

The solution is to a copy of the metadata data, and use the copy when calling the API.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/shell/audio.c [hunk 1]
```diff
@@ -822,18 +822,97 @@ static int cmd_qos(const struct shell *sh, size_t argc, char *argv[])
...
+
+	if (meta_str != NULL) {
+		uint16_t context;
+
+		context = strmeta(meta_str);
+		if (context == 0) {
+			return -ENOEXEC;
+		}
+
+		/* TODO: Check the type nad only overwrite the streaming context */
```
Comment (kruithofa) [inline subsys/bluetooth/shell/audio.c:882]:
```suggestion
		/* TODO: Check the type and only overwrite the streaming context */
```

File: subsys/bluetooth/shell/audio.c [hunk 2]
```diff
@@ -822,18 +822,97 @@ static int cmd_qos(const struct shell *sh, size_t argc, char *argv[])
...
+	if (argc > 1) {
+		err = handle_metadata_update(argv[1], &meta, &meta_count);
+
+		if (err != 0) {
+			shell_error(sh, "Unable to handle metadata update: %d",
+				    err);
+			return err;
+		}
+	} else {
+		err = handle_metadata_update(NULL, &meta, &meta_count);
```
Comment (kruithofa) [inline subsys/bluetooth/shell/audio.c:912]: erer is never used, isn't it then better to write
```suggestion
		(void)handle_metadata_update(NULL, &meta, &meta_count);
```

Comment (Thalley) [reply subsys/bluetooth/shell/audio.c:1106]: Moved the error check to after this one, so that it is handled.

**Review by kruithofa — APPROVED** [review]

File: subsys/bluetooth/shell/audio.c [hunk 3]
```diff
@@ -1022,18 +1022,96 @@ static int cmd_qos(const struct shell *sh, size_t argc, char *argv[])
...
 	if (default_stream == NULL) {
 		shell_error(sh, "Not connected");
 		return -ENOEXEC;
 	}
 
-	err = bt_audio_stream_enable(default_stream,
-				     default_preset->preset.codec.meta,
-				     default_preset->preset.codec.meta_count);
+	if (argc > 1) {
+		err = handle_metadata_update(argv[1], &meta, &meta_count);
```
Comment (MariuszSkamra) [inline subsys/bluetooth/shell/audio.c:1104]: The specification defines the context as bitmask. Is it possible to set multiple contexts? e.g. Media + Game.

Comment (Thalley) [reply subsys/bluetooth/shell/audio.c:1104]: Yes, we should extent this to support more than 1 argument, but that will be in a different PR. This is fixing an issue, and adding functionality can wait :)

**Review by MariuszSkamra — APPROVED** [review]

**Pull request merged**